### PR TITLE
metrics: add metrics agent to send metrics via telegraf

### PR DESCRIPTION
**Pull request by SanyaKochhar**
Signed-off-by: Sanya Kochhar <kochhars@microsoft.com>

**Description:**

* Adds a metric agent pod that helps send metrics to MDM
using Telegraf as a pipeline.
* prom-mdm-converter container helps convert Prometheus
metrics into statsd format for consumption by MDM.

**Testing done:**
- Tested that kube-inventory outputs match resource limits set on pods
- Tested setting metrics namespace as the region of extension installation
- Bats tests added for basic readiness of metrics pods

Follow up: metrics in a proxy environment

**Affected area**:

- New Functionality      [x]
- Documentation          [ ]
- Install                [ ]
- Networking             [ ]
- Metrics                [x]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

### CODE REVIEW

File: charts/osm-arc/templates/metrics-agent-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  template:
+    metadata:
+      labels:
+        app: osm-metrics-agent
+        {{- include "osm.arcLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ .Release.Name }}
+      containers:
+        - name: mdm
+          image: linuxgeneva-microsoft.azurecr.io/genevamdm:master_20220401.1
```
Comment (nshankar13) [inline charts/osm-arc/templates/metrics-agent-deployment.yaml:23]: I would add images as fields in "values.yaml" and then reference them here. like how osm-label does for the Azure Policy / alpine image.

File: charts/osm-arc/templates/metrics-agent-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+              value: /tmp/geneva_mdm/mdmconfig.json
+            - name: MDM_INPUT
+              value: statsd_udp,statsd_tcp
+            - name: MDM_LOG_LEVEL
+              value: "Info"
+          volumeMounts:
+            - name: mdm-config
+              mountPath: /tmp/geneva_mdm
+        - name: msi-adapter
+          image: mcr.microsoft.com/azurearck8s/msi-adapter:1.0.0
```
Comment (nshankar13) [inline charts/osm-arc/templates/metrics-agent-deployment.yaml:47]: Same as above.

File: charts/osm-arc/templates/metrics-config.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    [agent]
+      interval = "60s"
+      flush_interval = "10s"
+      metric_batch_size = 250
+      metric_buffer_limit = 1000
+
+    [[inputs.prometheus]]
+      monitor_kubernetes_pods = true
+      kubernetes_label_selector = "app.kubernetes.io/component=osm-controller"
+      monitor_kubernetes_pods_namespace = "arc-osm-system"
```
Comment (nshankar13) [inline charts/osm-arc/templates/metrics-config.yaml:25]: {{ Release.Namespace }} for consistency

File: charts/osm-arc/templates/metrics-agent-deployment.yaml (hunk 2 above)
Comment (SanyaKochhar) [reply charts/osm-arc/templates/metrics-agent-deployment.yaml:47]: same reasoning as above^

File: charts/osm-arc/templates/metrics-config.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: osm-telegraf-config
+  namespace: {{ .Release.Namespace }}
+data:
+  telegraf.conf: |-
+    [agent]
+      interval = "60s"
```
[the remark this answers is not in the document]
Comment (SanyaKochhar) [reply charts/osm-arc/templates/metrics-config.yaml:17]: same reasoning as above^

**Review by nshankar13 — APPROVED** [review]

**Review by trstringer — APPROVED** [review]

**Pull request merged**